src/Oceananigans.jl

```diff
@@ -153,6 +153,18 @@ Abstract supertype for models.
 """
 abstract type AbstractModel{TS} end
 
+"""
+    AbstractLagrangianParticles
+Abstract supertype for lagranigan particles.
+"""
+abstract type AbstractLagrangianParticles end
+
+update_particle_properties!(::Nothing, model, Δt) = nothing
+
+update_particle_properties!(model, Δt) = update_particle_properties!(model.particles, model, Δt)
+
+update_particle_properties!(particles::AbstractLagrangianParticles, model, Δt) = error("You need to define an `update_particle_properties!` method for $(typeof(particles))")

```

Comment:
We want to avoid new definitions as the top level, so we should remove this

Reply:
see https://github.com/CliMA/Oceananigans.jl/issues/3048

Reply:
Should I move the fallbacks into the top of the particles file? I guess maybe in the model file maybe? I think we need fallbacks for the abstract type before the model definition.

Reply:
Should I move the fallbacks into the top of the particles file? I guess maybe in the model file maybe? I think we need fallbacks for the abstract type before the model definition.

Reply:
I'm thinking we should move lagrangian particle stuff into `Models`. I don't want our work to conflict though, so let me know if you're ok with me working on this

Reply:
Okay yeah makes sense, you go for it!

Reply:
done, though we will see if tests pass.